vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -914,12 +939,18 @@ def _load_w13(self,
                                                  shard_size)
         # Narrow parameter and load.
         # w1, gate_proj: Load into first logical weight of w13.
-        if shard_id == "w1":
-            expert_data = expert_data.narrow(shard_dim, 0, shard_size)
         # w3, up_proj: Load into second logical weight of w13.
+        # The FlashInfer Cutlass fused MoE kernel expects the combined weights
+        # to be ordered as [w3, w1], unlike the standard [w1, w3] layout.
+        assert shard_id in ("w1", "w3")
+        switch_w13 = getattr(self.quant_method, 'load_up_proj_weight_first',
+                             False)
+        if (switch_w13 and shard_id == "w1") or (not switch_w13
+                                                 and shard_id == "w3"):
+            start = shard_size
```

Comment:
Could we put this switch logic in `process_weights_after_loading` in modelopt.py? Usually if we need to process the weights/scales after loading, we prefer to do it within the quant method if possible